### PR TITLE
Swap blueprinter out for active model serializer

### PR DESCRIPTION
**Pull request by BatedUrGonnaDie**
Blueprinter is actively being maintained, has better performance over AMS, and is quite nice to work with.  Some stats from my tests:
Response Time: 35s -> 28s (95 percentile)
Response Time: 29s -> 24s (mean time)
Memory Level Off: 1106 -> 939 (15% savings)

Unfortunately it does not pass the current view down to associations unless you explicitly pass it, which means there is a bit of copy pasting that I would've liked to put in the default view.  One other big gotcha was that if you put an incorrect view name it will silently default back to default instead of throwing an error which was a bit of a pain in the butt.

This should also fix a couple various 500's that I found while testing.

Removes the AMS block from #517 .

### CODE REVIEW

File: app/controllers/api/v3/games_controller.rb [hunk 1]
```diff
@@ -7,18 +7,18 @@ def index
...
 
   def show
-    render json: @game, serializer: Api::V3::GameSerializer, root: :game
+    render json: GameBlueprint.render(@game, view: :api_v3, root: :game, toplevel: :game)
   end
 
   private
 
   def set_game
-    @game = Game.joins(:srdc).includes(:categories).find_by(speedrun_dot_com_games: {shortname: params[:id]}) || Game.includes(:categories).find(params[:id])
```
Comment (glacials) [inline app/controllers/api/v3/games_controller.rb]: Any specific reason for removing these `includes` calls? Does Blueprinter handle it for us now?

**Comment by BatedUrGonnaDie** [comment]
Do you think future api version will share the same fields that are currently shared between v3 and v4?  If not then there will be even less fields in the default view to the point where it might not even be included into the new api.  Right now v3 is especially susceptible to changes since we don't have any schema matching like v4 does.  Fields could change that aren't checked or new fields could be added.

Also since view is not passed along to associations and has to be manually passed, it means that only the field calls can be put in the shared view.  I was also not able to find a way to override a field defined in an included view, which means that anything that has to change between versions needs to be kept out of the shared views.  The identifier as well will show up in EVERY view if it is defined inside of ANY view.  So if v4 defines an identifier, it will be present in v3 even if we call the v3 view and v3 does not include the v4 view.

Does any of this tilt the scales for you on the shared view front?

**Comment by glacials** [comment]
> The identifier as well will show up in EVERY view if it is defined inside of ANY view. So if v4 defines an identifier, it will be present in v3 even if we call the v3 view and v3 does not include the v4 view.

Wow ok that's insane. That definitely tips the scales for me. Why does this happen?

> Do you think future api version will share the same fields that are currently shared between v3 and v4? 

Yeah, that's very likely. However my personal viewpoint on bumping API versions has been gliding more towards the "we should almost never bump versions" end of the spectrum recently. I think we should stay on v4 for many years to come.

**Comment by BatedUrGonnaDie** [comment]
> Wow ok that's insane. That definitely tips the scales for me. Why does this happen?

I believe it is because the entire blueprint is parsed before anything actually is rendered.  Take the following code in the current run blueprint
```ruby
view :test do
  raise
end
```
will raise a runtime error when trying to render the blueprint, even though we are calling the `:api_v4` view and the `:test` view is not included in the current view.  I believe it just picks up an identifier statement which then it interprets for all views even if you are not using it in the current view.  Since the identifier is not required though, I was able to just make the `:id` column that would normally be the identifier a normal field in the blueprints in which it doesn't remain consistent across api versions.

**Comment by glacials** [comment]
🤔 That's still super weird and seems like a landmine. Are we sure Blueprinter views are meant to be used like this, to differentiate API versions? The only example I see is "standard" vs. "extended" views, which seems at odds with what we're trying to do, specify independent versions.

**Comment by BatedUrGonnaDie** [comment]
Uh, no I'm not sure if views were intended for this kind of use at all.  I was trying to condense the serializers down so that it would be one file per object, but I can see how it might be a landmine in the future for someone who doesn't know about some of these weird behaviors.  Maybe it is best to just have separate files for each version of the api then.

**Comment by glacials** [comment]
We could also just kill v3, it's been deprecated for 11 months now.

**Comment by BatedUrGonnaDie** [comment]
Doesn't Livesplit proper and SRDC still use v3?

**Comment by glacials** [comment]
Ah right, ok let's not kill it then O_o

**Comment by BatedUrGonnaDie** [comment]
I do think we should make some sort of progress towards killing off v3 since it does hold back other integrations (no gametime shown on SRDC splits).  Maybe we should add a depreciation notice to the v3 doc and add a kill date for it?  For this PR though if the plan is to remove v3 in the coming weeks/months then having separate files would probably be even better, because then we can just delete all the files related to v3 instead of having to go into the blueprints and remove just the v3 stuff.

**Comment by glacials** [comment]
Yeah that's a really good point. Okay, let's do that -- deprecation notice, and separate out into different files.

**Comment by BatedUrGonnaDie** [comment]
Alright, I'll extract into some kind of versioned layout and then fix that includes stuff in the controllers.

**Comment by BatedUrGonnaDie** [comment]
It does feel a lot less hacky having each api live in their own blueprints, and it removes pretty much all of the pitfalls that I was dodging around.

I did end up changing the output of the convert endpoint to what it's supposed to be outputting.  The docs say that it is supposed to have the output as the run endpoints, so I went ahead and updated it to reflect that (with the null id being the exception).  I also tried to fix most of the rubocop errors that were present in files that I touched.

**Pull request merged**